### PR TITLE
#28770 Register embeddings listener on init

### LINKED ISSUE #28770: dotAI: register EmbeddingContentListener
### Parent Issue

_No response_

### Problem Statement

The EmbeddingContentListener is not being registered after migrating the dotAI plugin into core. 

It was registered here in the plugin's activator.
https://github.com/dotCMS/com.dotcms.ai/blob/main/src/main/java/com/dotcms/ai/Activator.java#L188

This is important because it keeps the index in sync in push publish environments.

### Steps to Reproduce

1. Configure dotAI in the sender and receiver
2. Create an embeddings index for blogs on both.
3. Create and push a new blog
Error: only the embedding index in the sender will be updated while the one in the receiver should as well. 

### Acceptance Criteria

- [ ] Pushing a piece of content updates the embedding indices in both sender and receiver
- [ ] Push deleting a piece of content updates the embedding indices in both sender and receiver
- [ ] Integration Test

### dotCMS Version

24.04 LTS

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by dsilvam**
Register EmbeddingContentListener on init.

Fixes https://github.com/dotCMS/core/issues/28770

### CODE REVIEW

File: dotcms-integration/src/test/java/com/dotcms/ai/viewtool/AiTest.java [hunk 1]
```diff
@@ -84,6 +84,13 @@ private static Map<String, Secret> appConfigMap(final WireMockServer wireMockSer
...
                 AppKeys.IMAGE_SIZE.key,
-                Secret.builder().withType(Type.SELECT).withValue(IMAGE_SIZE.toCharArray()).build());
+                Secret.builder().withType(Type.SELECT).withValue(IMAGE_SIZE.toCharArray()).build(),
+
+                AppKeys.LISTENER_INDEXER.key,
+                Secret.builder()
+                        .withType(Type.STRING)
+                        .withValue("{\"default\":\"blog\"}".toCharArray())
+//                        .withValue("{\"default\":\"blog\",\"blogsOnly\":\"blog.body\"}".toCharArray())
+                        .build());
```
Comment (dsilvam) [inline dotcms-integration/src/test/java/com/dotcms/ai/viewtool/AiTest.java:94]: Why is this part of this PR?

File: dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java [hunk 2]
```diff
@@ -178,9 +186,8 @@ public List<Field> guessWhatFieldsToIndex(@NotNull Contentlet contentlet) {
         return contentlet.getContentType()
                 .fields()
                 .stream().filter(f -> !ignoreUrlMapFields.contains("{" + f.variable() + "}"))
-                .filter(f ->
-                        f.dataType().equals(DataTypes.LONG_TEXT)
-                ).collect(Collectors.toList());
+                .filter(f -> f.dataType().equals(DataTypes.LONG_TEXT))
+                .collect(Collectors.toList());
     }
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java:191]: Why including formatting changes?

File: dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java [hunk 3]
```diff
@@ -163,12 +164,19 @@ public List<Field> guessWhatFieldsToIndex(@NotNull Contentlet contentlet) {
...
                 .fields()
-                .stream().filter(f -> !ignoreUrlMapFields.contains("{" + f.variable() + "}"))
-                .filter(f -> f instanceof StoryBlockField || f instanceof WysiwygField || f instanceof BinaryField ||  f instanceof TextAreaField || f instanceof FileField)
+                .stream()
+                .filter(f -> !ignoreUrlMapFields.contains("{" + f.variable() + "}"))
+                .filter(f -> f instanceof StoryBlockField
+                        || f instanceof WysiwygField
+                        || f instanceof BinaryField
+                        || f instanceof TextAreaField
+                        || f instanceof FileField)
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java:179]: These are appreciated but again let's not include them as part of another chance because it makes difficult to understand/review the actual changes and inflates the PR.

File: dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java [hunk 4]
```diff
@@ -163,12 +164,19 @@ public List<Field> guessWhatFieldsToIndex(@NotNull Contentlet contentlet) {
             return List.of();
         }
 
-        final String ignoreUrlMapFields = (contentlet.getContentType().urlMapPattern() != null) ? contentlet.getContentType().urlMapPattern() : "";
+        final String ignoreUrlMapFields = org.apache.commons.lang3.StringUtils.defaultIfBlank(
+                contentlet.getContentType().urlMapPattern(),
+                StringPool.BLANK);
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java:169]: Shouldn't belong to this PR.

File: dotCMS/src/main/java/com/dotcms/ai/app/ConfigService.java [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 import com.dotcms.api.web.HttpServletRequestThreadLocal;
 import com.dotcms.security.apps.AppSecrets;
+import com.dotcms.security.apps.Secret;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/app/ConfigService.java:5]: Why a new import?

File: dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsRunner.java [hunk 6]
```diff
@@ -73,17 +73,22 @@ public void run() {
...
             if (buffer.toString().split("\\s+").length > 0) {
                 saveEmbedding(buffer.toString(), contentlet, indexName);
             }
         } catch (Exception e) {
             if (ConfigService.INSTANCE.config().getConfigBoolean(AppKeys.DEBUG_LOGGING)) {
-                Logger.warn(this.getClass(), "unable to embed content:" + contentlet.getIdentifier() + " error:" + e.getMessage(), e);
+                Logger.warn(
+                        this.getClass(),
+                        "unable to embed content:" + contentlet.getIdentifier() + " error:" + e.getMessage(),
+                        e);
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsRunner.java:86]: Not needed as part of this PR.

File: dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsRunner.java [hunk 7]
```diff
@@ -73,17 +73,22 @@ public void run() {
...
             } else {
-                Logger.warnAndDebug(this.getClass(), "unable to embed content:" + contentlet.getIdentifier() + " error:" + e.getMessage(), e);
+                Logger.warnAndDebug(
+                        this.getClass(),
+                        "unable to embed content:" + contentlet.getIdentifier() + " error:" + e.getMessage(),
+                        e);
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsRunner.java:91]: Not needed as part of this PR.

File: dotCMS/src/main/java/com/dotcms/ai/listener/EmbeddingContentListener.java [hunk 8]
```diff
@@ -109,47 +111,53 @@ JSONObject getConfigJson(Host host) {
...
             typesAndFields.entrySet()
                     .stream()
                     .filter(e -> contentType.equalsIgnoreCase(e.getKey()))
-                    .forEach(e ->
-                            EmbeddingsAPI.impl().generateEmbeddingsForContent(contentlet, e.getValue(), indexName)
-                    );
+                    .forEach(e -> EmbeddingsAPI.impl()
+                            .generateEmbeddingsForContent(
+                                    contentlet,
+                                    e.getValue(),
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/ai/listener/EmbeddingContentListener.java:132]: e is probably a wrong name

File: dotcms-integration/src/test/java/com/dotcms/ai/viewtool/AiTest.java (hunk 1 above)
Comment (victoralfaro-dotcms) [reply dotcms-integration/src/test/java/com/dotcms/ai/viewtool/AiTest.java:94]: Needed to provide the listener configuration. Removing commented code.

File: dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java (hunk 2 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java:191]: Rollbacked.

File: dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java (hunk 3 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java:179]: Rollbacked.

File: dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java (hunk 4 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/util/ContentToStringUtil.java:169]: Rollbacked.

File: dotCMS/src/main/java/com/dotcms/ai/app/ConfigService.java (hunk 5 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/app/ConfigService.java:5]: Removed.

File: dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsRunner.java (hunk 7 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsRunner.java:93]: Rollbacked.

File: dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsRunner.java (hunk 6 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsRunner.java]: Rollbacked.

File: dotCMS/src/main/java/com/dotcms/ai/listener/EmbeddingContentListener.java (hunk 8 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/listener/EmbeddingContentListener.java:132]: Renamed.

File: dotcms-integration/src/test/java/com/dotcms/datagen/TestDataUtils.java [hunk 9]
```diff
@@ -1248,6 +1260,23 @@ public static Contentlet getBlogContent(Boolean persist, long languageId,
         return getBlogContent(persist, languageId, contentTypeId, null);
     }
 
+    public static Contentlet getBlogContentWithEmbeddings(final boolean persist,
```
Comment (dsilvam) [inline dotcms-integration/src/test/java/com/dotcms/datagen/TestDataUtils.java:1263]: is "WithEmbeddings" the proper name? maybe "WithSEO" instead?

**Review by freddyDOTCMS — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]
